auction/src/lib.rs

```diff
@@ -98,7 +98,10 @@ decl_error! {
 
 impl<T: Trait> Module<T> {
 	fn _on_finalize(now: T::BlockNumber) {
-		for (auction_id, _) in <AuctionEndTime<T>>::drain(&now) {
+		for (block_number, auction_id, _) in <AuctionEndTime<T>>::drain() {
```

Comment:
this isn't right, it should be using `drain_prefix`

https://github.com/paritytech/substrate/blob/b096036932c46a0b9da543fb826be20224a45bc7/frame/support/src/storage/mod.rs#L239